servicetalk-http-router-predicate/src/main/java/io/servicetalk/http/router/predicate/HttpPredicateRouterBuilder.java

```diff
@@ -276,17 +275,13 @@ public RouteStarter thenRouteTo(final BlockingStreamingHttpService service) {
             return thenRouteTo0(adapterHolder.adaptor(), adapterHolder.serviceInvocationStrategy());
         }
 
-        private HttpExecutionStrategy serviceOffloads(final Object service) {
-            return null != strategy ? strategy :
-                    service instanceof ExecutionStrategyInfluencer ?
-                            HttpExecutionStrategy.from(((ExecutionStrategyInfluencer) service).requiredOffloads()) :
-                            defaultStrategy();
+        private HttpExecutionStrategy serviceOffloads(final HttpExecutionStrategyInfluencer service) {
+            return null != strategy ? strategy : service.requiredOffloads();
```

Comment:
👍

Reply:
While you are here: it looks like `InOrderRouter` has to implement `requireOffloads()` to return `offloadNone()`, doesn't it?

Reply:
I will take a look at this in a separate PR.